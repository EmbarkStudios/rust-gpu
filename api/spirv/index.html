<!DOCTYPE html><html lang="en"><head><meta charset="utf-8"><meta name="viewport" content="width=device-width, initial-scale=1.0"><meta name="generator" content="rustdoc"><meta name="description" content="The SPIR-V header."><meta name="keywords" content="rust, rustlang, rust-lang, spirv"><title>spirv - Rust</title><link rel="preload" as="font" type="font/woff2" crossorigin href="../SourceSerif4-Regular.ttf.woff2"><link rel="preload" as="font" type="font/woff2" crossorigin href="../FiraSans-Regular.woff2"><link rel="preload" as="font" type="font/woff2" crossorigin href="../FiraSans-Medium.woff2"><link rel="preload" as="font" type="font/woff2" crossorigin href="../SourceCodePro-Regular.ttf.woff2"><link rel="preload" as="font" type="font/woff2" crossorigin href="../SourceSerif4-Bold.ttf.woff2"><link rel="preload" as="font" type="font/woff2" crossorigin href="../SourceCodePro-Semibold.ttf.woff2"><link rel="stylesheet" type="text/css" href="../normalize.css"><link rel="stylesheet" type="text/css" href="../rustdoc.css" id="mainThemeStyle"><link rel="stylesheet" type="text/css" href="../ayu.css" disabled><link rel="stylesheet" type="text/css" href="../dark.css" disabled><link rel="stylesheet" type="text/css" href="../light.css" id="themeStyle"><script id="default-settings" ></script><script src="../storage.js"></script><script src="../crates.js"></script><script defer src="../main.js"></script>
    <noscript><link rel="stylesheet" href="../noscript.css"></noscript><link rel="alternate icon" type="image/png" href="../favicon-16x16.png"><link rel="alternate icon" type="image/png" href="../favicon-32x32.png"><link rel="icon" type="image/svg+xml" href="../favicon.svg"></head><body class="rustdoc mod crate"><!--[if lte IE 11]><div class="warning">This old browser is unsupported and will most likely display funky things.</div><![endif]--><nav class="mobile-topbar"><button class="sidebar-menu-toggle">&#9776;</button><a class="sidebar-logo" href="../spirv/index.html"><div class="logo-container"><img class="rust-logo" src="../rust-logo.svg" alt="logo"></div>
        </a><h2 class="location"></h2>
    </nav>
    <nav class="sidebar"><a class="sidebar-logo" href="../spirv/index.html"><div class="logo-container"><img class="rust-logo" src="../rust-logo.svg" alt="logo"></div>
        </a><h2 class="location"><a href="#">Crate spirv</a></h2><div class="sidebar-elems"><div class="block"><ul><li class="version">Version 0.2.0+1.5.4</li><li><a id="all-types" href="all.html">All Items</a></li></ul></div><section><div class="block"><ul><li><a href="#structs">Structs</a></li><li><a href="#enums">Enums</a></li><li><a href="#constants">Constants</a></li><li><a href="#types">Type Definitions</a></li></ul></div></section><div id="sidebar-vars" data-name="spirv" data-ty="mod" data-relpath=""></div><script defer src="sidebar-items.js"></script></div></nav><main><div class="width-limiter"><div class="sub-container"><a class="sub-logo-container" href="../spirv/index.html"><img class="rust-logo" src="../rust-logo.svg" alt="logo"></a><nav class="sub"><form class="search-form"><div class="search-container"><span></span><input class="search-input" name="search" autocomplete="off" spellcheck="false" placeholder="Click or press ‘S’ to search, ‘?’ for more options…" type="search"><button type="button" id="help-button" title="help">?</button><div id="settings-menu" tabindex="-1">
                                <a href="../settings.html" title="settings"><img width="22" height="22" alt="Change settings" src="../wheel.svg"></a></div>
                        </div></form></nav></div><section id="main-content" class="content"><div class="main-heading">
    <h1 class="fqn"><span class="in-band">Crate <a class="mod" href="#">spirv</a><button id="copy-path" onclick="copy_path(this)" title="Copy item path to clipboard"><img src="../clipboard.svg" width="19" height="18" alt="Copy item path"></button></span></h1><span class="out-of-band"><a class="srclink" href="../src/spirv/lib.rs.html#1-14">source</a> · <a id="toggle-all-docs" href="javascript:void(0)" title="collapse all docs">[<span class="inner">&#x2212;</span>]</a></span></div><details class="rustdoc-toggle top-doc" open><summary class="hideme"><span>Expand description</span></summary><div class="docblock"><p>The SPIR-V header.</p>
<p>This crate contains Rust definitions of all SPIR-V structs, enums,
and constants.</p>
<p>The version of this crate is the version of SPIR-V it contains.</p>
</div></details><h2 id="structs" class="small-section-header"><a href="#structs">Structs</a></h2>
<div class="item-table"><div class="item-row"><div class="item-left module-item"><a class="struct" href="struct.FPFastMathMode.html" title="spirv::FPFastMathMode struct">FPFastMathMode</a></div><div class="item-right docblock-short"><p>SPIR-V operand kind: <a href="https://www.khronos.org/registry/spir-v/specs/unified1/SPIRV.html#_a_id_fp_fast_math_mode_a_fp_fast_math_mode">FPFastMathMode</a></p>
</div></div><div class="item-row"><div class="item-left module-item"><a class="struct" href="struct.FragmentShadingRate.html" title="spirv::FragmentShadingRate struct">FragmentShadingRate</a></div><div class="item-right docblock-short"><p>SPIR-V operand kind: <a href="https://www.khronos.org/registry/spir-v/specs/unified1/SPIRV.html#_a_id_fragment_shading_rate_a_fragment_shading_rate">FragmentShadingRate</a></p>
</div></div><div class="item-row"><div class="item-left module-item"><a class="struct" href="struct.FunctionControl.html" title="spirv::FunctionControl struct">FunctionControl</a></div><div class="item-right docblock-short"><p>SPIR-V operand kind: <a href="https://www.khronos.org/registry/spir-v/specs/unified1/SPIRV.html#_a_id_function_control_a_function_control">FunctionControl</a></p>
</div></div><div class="item-row"><div class="item-left module-item"><a class="struct" href="struct.ImageOperands.html" title="spirv::ImageOperands struct">ImageOperands</a></div><div class="item-right docblock-short"><p>SPIR-V operand kind: <a href="https://www.khronos.org/registry/spir-v/specs/unified1/SPIRV.html#_a_id_image_operands_a_image_operands">ImageOperands</a></p>
</div></div><div class="item-row"><div class="item-left module-item"><a class="struct" href="struct.KernelProfilingInfo.html" title="spirv::KernelProfilingInfo struct">KernelProfilingInfo</a></div><div class="item-right docblock-short"><p>SPIR-V operand kind: <a href="https://www.khronos.org/registry/spir-v/specs/unified1/SPIRV.html#_a_id_kernel_profiling_info_a_kernel_profiling_info">KernelProfilingInfo</a></p>
</div></div><div class="item-row"><div class="item-left module-item"><a class="struct" href="struct.LoopControl.html" title="spirv::LoopControl struct">LoopControl</a></div><div class="item-right docblock-short"><p>SPIR-V operand kind: <a href="https://www.khronos.org/registry/spir-v/specs/unified1/SPIRV.html#_a_id_loop_control_a_loop_control">LoopControl</a></p>
</div></div><div class="item-row"><div class="item-left module-item"><a class="struct" href="struct.MemoryAccess.html" title="spirv::MemoryAccess struct">MemoryAccess</a></div><div class="item-right docblock-short"><p>SPIR-V operand kind: <a href="https://www.khronos.org/registry/spir-v/specs/unified1/SPIRV.html#_a_id_memory_access_a_memory_access">MemoryAccess</a></p>
</div></div><div class="item-row"><div class="item-left module-item"><a class="struct" href="struct.MemorySemantics.html" title="spirv::MemorySemantics struct">MemorySemantics</a></div><div class="item-right docblock-short"><p>SPIR-V operand kind: <a href="https://www.khronos.org/registry/spir-v/specs/unified1/SPIRV.html#_a_id_memory_semantics_a_memory_semantics">MemorySemantics</a></p>
</div></div><div class="item-row"><div class="item-left module-item"><a class="struct" href="struct.RayFlags.html" title="spirv::RayFlags struct">RayFlags</a></div><div class="item-right docblock-short"><p>SPIR-V operand kind: <a href="https://www.khronos.org/registry/spir-v/specs/unified1/SPIRV.html#_a_id_ray_flags_a_ray_flags">RayFlags</a></p>
</div></div><div class="item-row"><div class="item-left module-item"><a class="struct" href="struct.SelectionControl.html" title="spirv::SelectionControl struct">SelectionControl</a></div><div class="item-right docblock-short"><p>SPIR-V operand kind: <a href="https://www.khronos.org/registry/spir-v/specs/unified1/SPIRV.html#_a_id_selection_control_a_selection_control">SelectionControl</a></p>
</div></div></div><h2 id="enums" class="small-section-header"><a href="#enums">Enums</a></h2>
<div class="item-table"><div class="item-row"><div class="item-left module-item"><a class="enum" href="enum.AccessQualifier.html" title="spirv::AccessQualifier enum">AccessQualifier</a></div><div class="item-right docblock-short"><p>/// SPIR-V operand kind: <a href="https://www.khronos.org/registry/spir-v/specs/unified1/SPIRV.html#_a_id_access_qualifier_a_access_qualifier">AccessQualifier</a></p>
</div></div><div class="item-row"><div class="item-left module-item"><a class="enum" href="enum.AddressingModel.html" title="spirv::AddressingModel enum">AddressingModel</a></div><div class="item-right docblock-short"><p>/// SPIR-V operand kind: <a href="https://www.khronos.org/registry/spir-v/specs/unified1/SPIRV.html#_a_id_addressing_model_a_addressing_model">AddressingModel</a></p>
</div></div><div class="item-row"><div class="item-left module-item"><a class="enum" href="enum.BuiltIn.html" title="spirv::BuiltIn enum">BuiltIn</a></div><div class="item-right docblock-short"><p>/// SPIR-V operand kind: <a href="https://www.khronos.org/registry/spir-v/specs/unified1/SPIRV.html#_a_id_built_in_a_built_in">BuiltIn</a></p>
</div></div><div class="item-row"><div class="item-left module-item"><a class="enum" href="enum.CLOp.html" title="spirv::CLOp enum">CLOp</a></div><div class="item-right docblock-short"><p><a href="https://www.khronos.org/registry/spir-v/specs/unified1/OpenCL.ExtendedInstructionSet.100.html">OpenCL.std</a> extended instruction opcode</p>
</div></div><div class="item-row"><div class="item-left module-item"><a class="enum" href="enum.Capability.html" title="spirv::Capability enum">Capability</a></div><div class="item-right docblock-short"><p>/// SPIR-V operand kind: <a href="https://www.khronos.org/registry/spir-v/specs/unified1/SPIRV.html#_a_id_capability_a_capability">Capability</a></p>
</div></div><div class="item-row"><div class="item-left module-item"><a class="enum" href="enum.Decoration.html" title="spirv::Decoration enum">Decoration</a></div><div class="item-right docblock-short"><p>/// SPIR-V operand kind: <a href="https://www.khronos.org/registry/spir-v/specs/unified1/SPIRV.html#_a_id_decoration_a_decoration">Decoration</a></p>
</div></div><div class="item-row"><div class="item-left module-item"><a class="enum" href="enum.Dim.html" title="spirv::Dim enum">Dim</a></div><div class="item-right docblock-short"><p>/// SPIR-V operand kind: <a href="https://www.khronos.org/registry/spir-v/specs/unified1/SPIRV.html#_a_id_dim_a_dim">Dim</a></p>
</div></div><div class="item-row"><div class="item-left module-item"><a class="enum" href="enum.ExecutionMode.html" title="spirv::ExecutionMode enum">ExecutionMode</a></div><div class="item-right docblock-short"><p>/// SPIR-V operand kind: <a href="https://www.khronos.org/registry/spir-v/specs/unified1/SPIRV.html#_a_id_execution_mode_a_execution_mode">ExecutionMode</a></p>
</div></div><div class="item-row"><div class="item-left module-item"><a class="enum" href="enum.ExecutionModel.html" title="spirv::ExecutionModel enum">ExecutionModel</a></div><div class="item-right docblock-short"><p>/// SPIR-V operand kind: <a href="https://www.khronos.org/registry/spir-v/specs/unified1/SPIRV.html#_a_id_execution_model_a_execution_model">ExecutionModel</a></p>
</div></div><div class="item-row"><div class="item-left module-item"><a class="enum" href="enum.FPRoundingMode.html" title="spirv::FPRoundingMode enum">FPRoundingMode</a></div><div class="item-right docblock-short"><p>/// SPIR-V operand kind: <a href="https://www.khronos.org/registry/spir-v/specs/unified1/SPIRV.html#_a_id_fp_rounding_mode_a_fp_rounding_mode">FPRoundingMode</a></p>
</div></div><div class="item-row"><div class="item-left module-item"><a class="enum" href="enum.FunctionParameterAttribute.html" title="spirv::FunctionParameterAttribute enum">FunctionParameterAttribute</a></div><div class="item-right docblock-short"><p>/// SPIR-V operand kind: <a href="https://www.khronos.org/registry/spir-v/specs/unified1/SPIRV.html#_a_id_function_parameter_attribute_a_function_parameter_attribute">FunctionParameterAttribute</a></p>
</div></div><div class="item-row"><div class="item-left module-item"><a class="enum" href="enum.GLOp.html" title="spirv::GLOp enum">GLOp</a></div><div class="item-right docblock-short"><p><a href="https://www.khronos.org/registry/spir-v/specs/unified1/GLSL.std.450.html">GLSL.std.450</a> extended instruction opcode</p>
</div></div><div class="item-row"><div class="item-left module-item"><a class="enum" href="enum.GroupOperation.html" title="spirv::GroupOperation enum">GroupOperation</a></div><div class="item-right docblock-short"><p>/// SPIR-V operand kind: <a href="https://www.khronos.org/registry/spir-v/specs/unified1/SPIRV.html#_a_id_group_operation_a_group_operation">GroupOperation</a></p>
</div></div><div class="item-row"><div class="item-left module-item"><a class="enum" href="enum.ImageChannelDataType.html" title="spirv::ImageChannelDataType enum">ImageChannelDataType</a></div><div class="item-right docblock-short"><p>/// SPIR-V operand kind: <a href="https://www.khronos.org/registry/spir-v/specs/unified1/SPIRV.html#_a_id_image_channel_data_type_a_image_channel_data_type">ImageChannelDataType</a></p>
</div></div><div class="item-row"><div class="item-left module-item"><a class="enum" href="enum.ImageChannelOrder.html" title="spirv::ImageChannelOrder enum">ImageChannelOrder</a></div><div class="item-right docblock-short"><p>/// SPIR-V operand kind: <a href="https://www.khronos.org/registry/spir-v/specs/unified1/SPIRV.html#_a_id_image_channel_order_a_image_channel_order">ImageChannelOrder</a></p>
</div></div><div class="item-row"><div class="item-left module-item"><a class="enum" href="enum.ImageFormat.html" title="spirv::ImageFormat enum">ImageFormat</a></div><div class="item-right docblock-short"><p>/// SPIR-V operand kind: <a href="https://www.khronos.org/registry/spir-v/specs/unified1/SPIRV.html#_a_id_image_format_a_image_format">ImageFormat</a></p>
</div></div><div class="item-row"><div class="item-left module-item"><a class="enum" href="enum.KernelEnqueueFlags.html" title="spirv::KernelEnqueueFlags enum">KernelEnqueueFlags</a></div><div class="item-right docblock-short"><p>/// SPIR-V operand kind: <a href="https://www.khronos.org/registry/spir-v/specs/unified1/SPIRV.html#_a_id_kernel_enqueue_flags_a_kernel_enqueue_flags">KernelEnqueueFlags</a></p>
</div></div><div class="item-row"><div class="item-left module-item"><a class="enum" href="enum.LinkageType.html" title="spirv::LinkageType enum">LinkageType</a></div><div class="item-right docblock-short"><p>/// SPIR-V operand kind: <a href="https://www.khronos.org/registry/spir-v/specs/unified1/SPIRV.html#_a_id_linkage_type_a_linkage_type">LinkageType</a></p>
</div></div><div class="item-row"><div class="item-left module-item"><a class="enum" href="enum.MemoryModel.html" title="spirv::MemoryModel enum">MemoryModel</a></div><div class="item-right docblock-short"><p>/// SPIR-V operand kind: <a href="https://www.khronos.org/registry/spir-v/specs/unified1/SPIRV.html#_a_id_memory_model_a_memory_model">MemoryModel</a></p>
</div></div><div class="item-row"><div class="item-left module-item"><a class="enum" href="enum.Op.html" title="spirv::Op enum">Op</a></div><div class="item-right docblock-short"><p>SPIR-V <a href="https://www.khronos.org/registry/spir-v/specs/unified1/SPIRV.html#_a_id_instructions_a_instructions">instructions</a> opcodes</p>
</div></div><div class="item-row"><div class="item-left module-item"><a class="enum" href="enum.RayQueryCandidateIntersectionType.html" title="spirv::RayQueryCandidateIntersectionType enum">RayQueryCandidateIntersectionType</a></div><div class="item-right docblock-short"><p>/// SPIR-V operand kind: <a href="https://www.khronos.org/registry/spir-v/specs/unified1/SPIRV.html#_a_id_ray_query_candidate_intersection_type_a_ray_query_candidate_intersection_type">RayQueryCandidateIntersectionType</a></p>
</div></div><div class="item-row"><div class="item-left module-item"><a class="enum" href="enum.RayQueryCommittedIntersectionType.html" title="spirv::RayQueryCommittedIntersectionType enum">RayQueryCommittedIntersectionType</a></div><div class="item-right docblock-short"><p>/// SPIR-V operand kind: <a href="https://www.khronos.org/registry/spir-v/specs/unified1/SPIRV.html#_a_id_ray_query_committed_intersection_type_a_ray_query_committed_intersection_type">RayQueryCommittedIntersectionType</a></p>
</div></div><div class="item-row"><div class="item-left module-item"><a class="enum" href="enum.RayQueryIntersection.html" title="spirv::RayQueryIntersection enum">RayQueryIntersection</a></div><div class="item-right docblock-short"><p>/// SPIR-V operand kind: <a href="https://www.khronos.org/registry/spir-v/specs/unified1/SPIRV.html#_a_id_ray_query_intersection_a_ray_query_intersection">RayQueryIntersection</a></p>
</div></div><div class="item-row"><div class="item-left module-item"><a class="enum" href="enum.SamplerAddressingMode.html" title="spirv::SamplerAddressingMode enum">SamplerAddressingMode</a></div><div class="item-right docblock-short"><p>/// SPIR-V operand kind: <a href="https://www.khronos.org/registry/spir-v/specs/unified1/SPIRV.html#_a_id_sampler_addressing_mode_a_sampler_addressing_mode">SamplerAddressingMode</a></p>
</div></div><div class="item-row"><div class="item-left module-item"><a class="enum" href="enum.SamplerFilterMode.html" title="spirv::SamplerFilterMode enum">SamplerFilterMode</a></div><div class="item-right docblock-short"><p>/// SPIR-V operand kind: <a href="https://www.khronos.org/registry/spir-v/specs/unified1/SPIRV.html#_a_id_sampler_filter_mode_a_sampler_filter_mode">SamplerFilterMode</a></p>
</div></div><div class="item-row"><div class="item-left module-item"><a class="enum" href="enum.Scope.html" title="spirv::Scope enum">Scope</a></div><div class="item-right docblock-short"><p>/// SPIR-V operand kind: <a href="https://www.khronos.org/registry/spir-v/specs/unified1/SPIRV.html#_a_id_scope_a_scope">Scope</a></p>
</div></div><div class="item-row"><div class="item-left module-item"><a class="enum" href="enum.SourceLanguage.html" title="spirv::SourceLanguage enum">SourceLanguage</a></div><div class="item-right docblock-short"><p>/// SPIR-V operand kind: <a href="https://www.khronos.org/registry/spir-v/specs/unified1/SPIRV.html#_a_id_source_language_a_source_language">SourceLanguage</a></p>
</div></div><div class="item-row"><div class="item-left module-item"><a class="enum" href="enum.StorageClass.html" title="spirv::StorageClass enum">StorageClass</a></div><div class="item-right docblock-short"><p>/// SPIR-V operand kind: <a href="https://www.khronos.org/registry/spir-v/specs/unified1/SPIRV.html#_a_id_storage_class_a_storage_class">StorageClass</a></p>
</div></div></div><h2 id="constants" class="small-section-header"><a href="#constants">Constants</a></h2>
<div class="item-table"><div class="item-row"><div class="item-left module-item"><a class="constant" href="constant.MAGIC_NUMBER.html" title="spirv::MAGIC_NUMBER constant">MAGIC_NUMBER</a></div><div class="item-right docblock-short"></div></div><div class="item-row"><div class="item-left module-item"><a class="constant" href="constant.MAJOR_VERSION.html" title="spirv::MAJOR_VERSION constant">MAJOR_VERSION</a></div><div class="item-right docblock-short"></div></div><div class="item-row"><div class="item-left module-item"><a class="constant" href="constant.MINOR_VERSION.html" title="spirv::MINOR_VERSION constant">MINOR_VERSION</a></div><div class="item-right docblock-short"></div></div><div class="item-row"><div class="item-left module-item"><a class="constant" href="constant.REVISION.html" title="spirv::REVISION constant">REVISION</a></div><div class="item-right docblock-short"></div></div></div><h2 id="types" class="small-section-header"><a href="#types">Type Definitions</a></h2>
<div class="item-table"><div class="item-row"><div class="item-left module-item"><a class="type" href="type.Word.html" title="spirv::Word type">Word</a></div><div class="item-right docblock-short"></div></div></div></section></div></main><div id="rustdoc-vars" data-root-path="../" data-current-crate="spirv" data-themes="ayu,dark,light" data-resource-suffix="" data-rustdoc-version="1.63.0-nightly (546c826f0 2022-06-12)" ></div>
</body></html>