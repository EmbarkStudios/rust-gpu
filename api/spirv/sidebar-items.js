window.SIDEBAR_ITEMS = {"constant":["MAGIC_NUMBER","MAJOR_VERSION","MINOR_VERSION","REVISION"],"enum":["AccessQualifier","AddressingModel","BuiltIn","CLOp","Capability","Decoration","Dim","ExecutionMode","ExecutionModel","FPRoundingMode","FunctionParameterAttribute","GLOp","GroupOperation","ImageChannelDataType","ImageChannelOrder","ImageFormat","KernelEnqueueFlags","LinkageType","MemoryModel","Op","RayQueryCandidateIntersectionType","RayQueryCommittedIntersectionType","RayQueryIntersection","SamplerAddressingMode","SamplerFilterMode","Scope","SourceLanguage","StorageClass"],"struct":["FPFastMathMode","FragmentShadingRate","FunctionControl","ImageOperands","KernelProfilingInfo","LoopControl","MemoryAccess","MemorySemantics","RayFlags","SelectionControl"],"type":["Word"]};