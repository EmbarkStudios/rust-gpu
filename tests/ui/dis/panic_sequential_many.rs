#![crate_name = "panic_sequential_many"]

// Test a long sequence of conditional panics, which has historically generated
// very nested structured control-flow (instead of a single merged chain).

// build-pass
// compile-flags: -C target-feature=+ext:SPV_KHR_non_semantic_info
// compile-flags: -C llvm-args=--abort-strategy=debug-printf
// compile-flags: -C llvm-args=--disassemble

// FIXME(eddyb) consider using such replacements also for dealing
// with `OpLine` changing all the time (esp. in libcore functions).
//
// normalize-stderr-test "; (SPIR-V|Generator: rspirv|Version: 1\.\d+|Bound: \d+)\n" -> ""
// normalize-stderr-test "OpCapability VulkanMemoryModel\n" -> ""
// normalize-stderr-test "OpSource .*\n" -> ""
// normalize-stderr-test "OpExtension .SPV_KHR_vulkan_memory_model.\n" -> ""
// normalize-stderr-test "OpMemoryModel Logical Vulkan" -> "OpMemoryModel Logical Simple"

// FIXME(eddyb) handle this one in the test runner.
// normalize-stderr-test "\S*/lib/rustlib/" -> "$$SYSROOT/lib/rustlib/"

use spirv_std::spirv;

#[spirv(fragment)]
pub fn main(#[spirv(flat)] x: u32, #[spirv(flat)] y: u32, o: &mut u32) {
    // HACK(eddyb) this might stop working if the checks get optimized out,
    // after the first `y != 0` (which dominates the rest of the function).
    *o = x / y / y / y / y / y / y / y / y / y / y / y / y / y / y / y / y / y;
}
