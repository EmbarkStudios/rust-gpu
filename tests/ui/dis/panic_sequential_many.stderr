OpCapability Shader
OpCapability Float64
OpCapability Int64
OpCapability Int16
OpCapability Int8
OpCapability ShaderClockKHR
OpExtension "SPV_KHR_non_semantic_info"
OpExtension "SPV_KHR_shader_clock"
%1 = OpExtInstImport "NonSemantic.DebugPrintf"
OpMemoryModel Logical Simple
OpEntryPoint Fragment %2 "main" %3 %4 %5
OpExecutionMode %2 OriginUpperLeft
%6 = OpString "/n[Rust panicked at $DIR/panic_sequential_many.rs:29:10]/n attempt to divide by zero/n      in main()/n"
%7 = OpString "$DIR/panic_sequential_many.rs"
OpName %3 "x"
OpName %4 "y"
OpName %5 "o"
OpDecorate %3 Flat
OpDecorate %3 Location 0
OpDecorate %4 Flat
OpDecorate %4 Location 1
OpDecorate %5 Location 0
%8 = OpTypeInt 32 0
%9 = OpTypePointer Input %8
%10 = OpTypePointer Output %8
%11 = OpTypeVoid
%12 = OpTypeFunction %11
%3 = OpVariable  %9  Input
%4 = OpVariable  %9  Input
%13 = OpTypeBool
%14 = OpConstant  %8  0
%5 = OpVariable  %10  Output
%2 = OpFunction  %11  None %12
%15 = OpLabel
OpLine %7 26 12
%16 = OpLoad  %8  %3
OpLine %7 26 35
%17 = OpLoad  %8  %4
OpLine %7 29 9
%18 = OpIEqual  %13  %17 %14
OpNoLine
OpSelectionMerge %19 None
OpBranchConditional %18 %20 %21
%20 = OpLabel
OpLine %7 29 9
%22 = OpExtInst  %11  %1 1 %6
OpNoLine
OpBranch %19
%21 = OpLabel
OpLine %7 29 9
%23 = OpUDiv  %8  %16 %17
%24 = OpIEqual  %13  %17 %14
OpNoLine
OpSelectionMerge %25 None
OpBranchConditional %24 %26 %27
%26 = OpLabel
OpLine %7 29 9
%28 = OpExtInst  %11  %1 1 %6
OpNoLine
OpBranch %25
%27 = OpLabel
OpLine %7 29 9
%29 = OpUDiv  %8  %23 %17
%30 = OpIEqual  %13  %17 %14
OpNoLine
OpSelectionMerge %31 None
OpBranchConditional %30 %32 %33
%32 = OpLabel
OpLine %7 29 9
%34 = OpExtInst  %11  %1 1 %6
OpNoLine
OpBranch %31
%33 = OpLabel
OpLine %7 29 9
%35 = OpUDiv  %8  %29 %17
%36 = OpIEqual  %13  %17 %14
OpNoLine
OpSelectionMerge %37 None
OpBranchConditional %36 %38 %39
%38 = OpLabel
OpLine %7 29 9
%40 = OpExtInst  %11  %1 1 %6
OpNoLine
OpBranch %37
%39 = OpLabel
OpLine %7 29 9
%41 = OpUDiv  %8  %35 %17
%42 = OpIEqual  %13  %17 %14
OpNoLine
OpSelectionMerge %43 None
OpBranchConditional %42 %44 %45
%44 = OpLabel
OpLine %7 29 9
%46 = OpExtInst  %11  %1 1 %6
OpNoLine
OpBranch %43
%45 = OpLabel
OpLine %7 29 9
%47 = OpUDiv  %8  %41 %17
%48 = OpIEqual  %13  %17 %14
OpNoLine
OpSelectionMerge %49 None
OpBranchConditional %48 %50 %51
%50 = OpLabel
OpLine %7 29 9
%52 = OpExtInst  %11  %1 1 %6
OpNoLine
OpBranch %49
%51 = OpLabel
OpLine %7 29 9
%53 = OpUDiv  %8  %47 %17
%54 = OpIEqual  %13  %17 %14
OpNoLine
OpSelectionMerge %55 None
OpBranchConditional %54 %56 %57
%56 = OpLabel
OpLine %7 29 9
%58 = OpExtInst  %11  %1 1 %6
OpNoLine
OpBranch %55
%57 = OpLabel
OpLine %7 29 9
%59 = OpUDiv  %8  %53 %17
%60 = OpIEqual  %13  %17 %14
OpNoLine
OpSelectionMerge %61 None
OpBranchConditional %60 %62 %63
%62 = OpLabel
OpLine %7 29 9
%64 = OpExtInst  %11  %1 1 %6
OpNoLine
OpBranch %61
%63 = OpLabel
OpLine %7 29 9
%65 = OpUDiv  %8  %59 %17
%66 = OpIEqual  %13  %17 %14
OpNoLine
OpSelectionMerge %67 None
OpBranchConditional %66 %68 %69
%68 = OpLabel
OpLine %7 29 9
%70 = OpExtInst  %11  %1 1 %6
OpNoLine
OpBranch %67
%69 = OpLabel
OpLine %7 29 9
%71 = OpUDiv  %8  %65 %17
%72 = OpIEqual  %13  %17 %14
OpNoLine
OpSelectionMerge %73 None
OpBranchConditional %72 %74 %75
%74 = OpLabel
OpLine %7 29 9
%76 = OpExtInst  %11  %1 1 %6
OpNoLine
OpBranch %73
%75 = OpLabel
OpLine %7 29 9
%77 = OpUDiv  %8  %71 %17
%78 = OpIEqual  %13  %17 %14
OpNoLine
OpSelectionMerge %79 None
OpBranchConditional %78 %80 %81
%80 = OpLabel
OpLine %7 29 9
%82 = OpExtInst  %11  %1 1 %6
OpNoLine
OpBranch %79
%81 = OpLabel
OpLine %7 29 9
%83 = OpUDiv  %8  %77 %17
%84 = OpIEqual  %13  %17 %14
OpNoLine
OpSelectionMerge %85 None
OpBranchConditional %84 %86 %87
%86 = OpLabel
OpLine %7 29 9
%88 = OpExtInst  %11  %1 1 %6
OpNoLine
OpBranch %85
%87 = OpLabel
OpLine %7 29 9
%89 = OpUDiv  %8  %83 %17
%90 = OpIEqual  %13  %17 %14
OpNoLine
OpSelectionMerge %91 None
OpBranchConditional %90 %92 %93
%92 = OpLabel
OpLine %7 29 9
%94 = OpExtInst  %11  %1 1 %6
OpNoLine
OpBranch %91
%93 = OpLabel
OpLine %7 29 9
%95 = OpUDiv  %8  %89 %17
%96 = OpIEqual  %13  %17 %14
OpNoLine
OpSelectionMerge %97 None
OpBranchConditional %96 %98 %99
%98 = OpLabel
OpLine %7 29 9
%100 = OpExtInst  %11  %1 1 %6
OpNoLine
OpBranch %97
%99 = OpLabel
OpLine %7 29 9
%101 = OpUDiv  %8  %95 %17
%102 = OpIEqual  %13  %17 %14
OpNoLine
OpSelectionMerge %103 None
OpBranchConditional %102 %104 %105
%104 = OpLabel
OpLine %7 29 9
%106 = OpExtInst  %11  %1 1 %6
OpNoLine
OpBranch %103
%105 = OpLabel
OpLine %7 29 9
%107 = OpUDiv  %8  %101 %17
%108 = OpIEqual  %13  %17 %14
OpNoLine
OpSelectionMerge %109 None
OpBranchConditional %108 %110 %111
%110 = OpLabel
OpLine %7 29 9
%112 = OpExtInst  %11  %1 1 %6
OpNoLine
OpBranch %109
%111 = OpLabel
OpLine %7 29 9
%113 = OpUDiv  %8  %107 %17
%114 = OpIEqual  %13  %17 %14
OpNoLine
OpSelectionMerge %115 None
OpBranchConditional %114 %116 %117
%116 = OpLabel
OpLine %7 29 9
%118 = OpExtInst  %11  %1 1 %6
OpNoLine
OpBranch %115
%117 = OpLabel
OpLine %7 29 4
%119 = OpUDiv  %8  %113 %17
OpStore %5 %119
OpNoLine
OpBranch %115
%115 = OpLabel
OpBranch %109
%109 = OpLabel
OpBranch %103
%103 = OpLabel
OpBranch %97
%97 = OpLabel
OpBranch %91
%91 = OpLabel
OpBranch %85
%85 = OpLabel
OpBranch %79
%79 = OpLabel
OpBranch %73
%73 = OpLabel
OpBranch %67
%67 = OpLabel
OpBranch %61
%61 = OpLabel
OpBranch %55
%55 = OpLabel
OpBranch %49
%49 = OpLabel
OpBranch %43
%43 = OpLabel
OpBranch %37
%37 = OpLabel
OpBranch %31
%31 = OpLabel
OpBranch %25
%25 = OpLabel
OpBranch %19
%19 = OpLabel
OpReturn
OpFunctionEnd
